pkg/app/piped/executor/ecs/ecs.go

```diff
@@ -144,25 +136,34 @@ func build(ctx context.Context, in *executor.Input, client provider.Client, task
 			return false
 		}
 	}
-	// service.TaskDefinition = td.TaskDefinitionArn
-	// service.LaunchType = serviceDefinition.LaunchType
-	// service.LoadBalancers = serviceDefinition.LoadBalancers
-
-	// Create a task set in the specified cluster and service and routing traffic to that task set.
-	// This is used when a service uses the EXTERNAL deployment controller type.
-	// For more information, see Amazon ECS Deployment Types
-	// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-types.html
-	// Note: The deployment controller type can also get from types.Service but this field is omitted if the service is using the ECS
-	// deployment controller type, so that we should use serviceDefinition.DeploymentController instead.
-	if serviceDefinition.DeploymentController.Type == types.DeploymentControllerTypeExternal {
-		taskSet, err := client.CreateTaskSet(ctx, *service, *td, 100)
-		if err != nil {
-			in.LogPersister.Errorf("Failed to create ECS task set %s: %v", *serviceDefinition.ServiceName, err)
-			return false
-		}
 
-		if _, err = client.UpdateServicePrimaryTaskSet(ctx, *service, *taskSet); err != nil {
-			in.LogPersister.Errorf("Failed to update service primary ECS task set %s: %v", *serviceDefinition.ServiceName, err)
+	// hack: Set this two value to enable create TaskSet (those values are ignored on Update/Create Service).
+	service.LaunchType = serviceDefinition.LaunchType
+	service.NetworkConfiguration = serviceDefinition.NetworkConfiguration
+
+	// Get current PRIMARY task set.
+	prevPrimaryTaskSet, err := client.GetPrimaryTaskSet(ctx, *service)
+	if err != nil {
+		in.LogPersister.Errorf("Failed to create ECS task set %s: %v", *serviceDefinition.ServiceName, err)
+	}
```

Comment:
nit: I feel it more readable to move this part to L162 as related things get close together.

Reply:
In that case, we will get the newly promoted (to primary) task set since we called `UpdateServicePrimaryTaskSet` at L158 👀

Reply:
we will 👍 
makes sense, never mind